vsintegration/src/FSharp.Editor/Diagnostics/SimplifyNameDiagnosticAnalyzer.fs

```diff
@@ -17,15 +17,18 @@ open Microsoft.FSharp.Compiler.SourceCodeServices
 
 open Microsoft.VisualStudio.FSharp.LanguageService
 
-type private LineHash = int
+type private TextVersionHash = int
 
-//[<DiagnosticAnalyzer(FSharpCommonConstants.FSharpLanguageName)>]
+// TODO Turn it on when user settings dialog is ready to switch it on and off.
+// [<DiagnosticAnalyzer(FSharpCommonConstants.FSharpLanguageName)>]
 type internal SimplifyNameDiagnosticAnalyzer() =
     inherit DocumentDiagnosticAnalyzer()
     
     let getProjectInfoManager (document: Document) = document.Project.Solution.Workspace.Services.GetService<FSharpCheckerWorkspaceService>().ProjectInfoManager
     let getChecker (document: Document) = document.Project.Solution.Workspace.Services.GetService<FSharpCheckerWorkspaceService>().Checker
     let getPlidLength (plid: string list) = (plid |> List.sumBy String.length) + plid.Length
+    static let cache = ConditionalWeakTable<DocumentId, TextVersionHash * ImmutableArray<Diagnostic>>()
+    static let guard = new SemaphoreSlim(1)
```

Comment:
That's this lock for? Please add a big comment on this to explain why we need a lock here, thanks